packages/cli/src/printer.ts

```diff
@@ -0,0 +1,7 @@
+import { LogLevel, Printer } from '@aws-amplify/cli-core';
+
+const minimumLogLevel = process.argv.includes('--debug')
+  ? LogLevel.DEBUG
+  : LogLevel.INFO;
+
+export const printer = new Printer(minimumLogLevel);
```

Comment:
I'm on the fence about whether we should construct this here and import it as needed, or figure out a way to construct it in the main parser and inject it via ctors into the commands: https://github.com/aws-amplify/amplify-backend/blob/main/packages/cli/src/main_parser_factory.ts

At the very least, we should add the `--debug` flag to the main parser so it shows up in `--help`

@sobolk any thoughts here?

Reply:
I was trying to do this and it become a bit of nightmare trying to refactor all the changes. This will force us to attempt parse twice.

Reply:
Okay, I think I'm good with this approach for now. We should still add the `--debug` flag to the main parser though (even if it's not used for anything within the yargs process) and leave a comment pointing to where it is used here.